osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -339,26 +360,53 @@ private void updateCustomisationVisualState()
             TopLevelContent.MoveToY(-modAreaHeight, transition_duration, Easing.InOutCubic);
         }
 
-        private void updateSelectionFromBindable()
+        /// <summary>
+        /// This flag helps to determine the source of changes to <see cref="SelectedMods"/>.
+        /// If the value is false, then <see cref="SelectedMods"/> are changing due to a user selection on the UI.
+        /// If the value is true, then <see cref="SelectedMods"/> are changing due to an external <see cref="SelectedMods"/> change.
+        /// </summary>
+        private bool externalSelectionUpdateInProgress;
```

Comment:
Minor nit but the xmldoc isn't really true at all times. Internal selection changes `SelectedMods`, which triggers `updateFromExternalSelection()` which changes this to `true`.